app/scripts/modules/core/src/pipeline/config/preconditions/types/stageStatus/StageStatusPreconditionConfig.tsx

```diff
@@ -0,0 +1,83 @@
+import { module } from 'angular';
+
+import * as React from 'react';
+import { react2angular } from 'react2angular';
+import { Option } from 'react-select';
+import { get } from 'lodash';
+
+import { FormField, IStage, ReactSelectInput } from '@spinnaker/core';
+
+interface IStageStatusPreconditionConfigProps {
+  preconditionContext: any;
+  upstreamStages: IStage[];
+  updatePreconditionContext: (context: any) => void;
+}
+
+// todo(mneterval): are there user stories to justify adding additional status options?
+const STATUS_OPTIONS: Array<Option<string>> = [
+  {
+    label: 'Succeeded',
+    value: 'SUCCEEDED',
+  },
+  {
+    label: 'Terminal',
+    value: 'TERMINAL',
+  },
+];
+
+export function StageStatusPreconditionConfig({
+  preconditionContext,
+  upstreamStages,
+  updatePreconditionContext,
+}: IStageStatusPreconditionConfigProps) {
+  return (
+    <div className="form-group row">
+      <FormField
+        input={inputProps => (
+          <ReactSelectInput
+            {...inputProps}
+            clearable={false}
+            options={upstreamStages.map(stage => ({ value: stage.name, label: stage.name }))}
```

Comment:
_supermicro_ nit: could be destructed  `map(({ name }) => ({ value: name, label: name }))`